pallets/dmp-queue/src/migration.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A module that is responsible for migration of storage.
+
+use crate::{Config, Pallet, Store};
+use frame_support::{pallet_prelude::*, traits::StorageVersion, weights::{constants::WEIGHT_PER_MILLIS, Weight}};
+use xcm::latest::Weight as XcmWeight;
+
+/// The current storage version.
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+/// Migrates the pallet storage to the most recent version, checking and setting the
+/// `StorageVersion`.
+pub fn migrate_to_latest<T: Config>() -> Weight {
+	let mut weight = Weight::zero();
+
+	if StorageVersion::get::<Pallet<T>>() == 0 {
+		weight += migrate_to_v1::<T>();
+		StorageVersion::new(1).put::<Pallet<T>>();
+	}
+
+	weight
+}
+
+mod v0 {
+    use super::*;
+    use codec::{Decode, Encode};
+
+    #[derive(Decode, Encode, Debug)]
+    pub struct ConfigData {
+        pub max_individual: XcmWeight,
+    }
+
+    impl Default for ConfigData {
+        fn default() -> Self {
+            ConfigData {
+                max_individual: 10u64 * WEIGHT_PER_MILLIS.ref_time(),
+            }
+        }
+    }
+}
+
+/// Migrates `QueueConfigData` from v1 (using only reference time weights) to v2 (with
+/// 2D weights).
+///
+/// NOTE: Only use this function if you know what you're doing. Default to using
+/// `migrate_to_latest`.
```

Comment:
maybe an additional sentence about the scenario where one would _want_ to call this just as a sanity check for people who know what they're doing ;)

Reply:
Honestly, the situation where I think it would make sense to call `migrate_to_v1` directly is during unit testing, otherwise I really don't see why anyone would call this method on its own.